src/java.net.http/share/classes/jdk/internal/net/http/FilterFactory.java

```diff
@@ -25,20 +25,20 @@
 
 package jdk.internal.net.http;
 
-import java.util.LinkedList;
+import java.util.ArrayList;
 import java.util.List;
 
 class FilterFactory {
 
     // Strictly-ordered list of filters.
-    final LinkedList<Class<? extends HeaderFilter>> filterClasses = new LinkedList<>();
+    final List<Class<? extends HeaderFilter>> filterClasses = new ArrayList<>(3);
 
     public void addFilter(Class<? extends HeaderFilter> type) {
         filterClasses.add(type);
     }
 
-    LinkedList<HeaderFilter> getFilterChain() {
-        LinkedList<HeaderFilter> l = new LinkedList<>();
+    List<HeaderFilter> getFilterChain() {
```

Comment:
I was thinking about the ways how FilterChain could be simplified. I played around with the `List<Supplier<HeaderFilter>>` instead of `List<Class<? extends HeaderFilter>>` and it seems that such change might improve the FilterFactory: 
```
    List<HeaderFilter> getFilterChain() {
        return filterSuppliers.stream()
                .map(Supplier::get)
                .toList();
    }
```
There is no constructor reflective access anymore, hence no exception handling.
Drawback here would be that HttpClientImpl in the `initFilters` will need to add suppliers vs classes i.e. `filters.addFilter(AuthenticationFilter::new);` vs `addFilter(AuthenticationFilter.class);`.

I think FilterFactory looks a bit simpler this way. 
What do you think, should we go ahead with the change?